File: js-ts/O/after/electrical-protection/Strategy/FuseStrategy.ts
import { IProtectionStrategy } from "./IProtectionStrategy.ts";

export class FuseStrategy implements IProtectionStrategy {
    public calcIn(In: number): number {
        return In + 3;
    }

    public calcBreakingCapacity(): number {
        return 3;
    }

    public checkProtection(): boolean {
        return true;
    }
}

File: csharp/O/After/Strategy/FuseStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace After.Strategy;

internal class FuseStrategy : IProtectionStrategy
{
    public double CalcBreakingCapacity()
    {
        return 2;
    }

    public double CalcIn()
    {
        return 2;
    }

    public bool CheckProtection()
    {
        return true;
    }
}
